app/src/main/java/swati4star/createpdf/util/ExtractImages.java

```diff
@@ -23,25 +25,51 @@
     private ExtractImagesListener mExtractImagesListener;
     private int mImagesCount = 0;
     private ArrayList<String> mOutputFilePaths;
+    private PDFEncryptionUtility mPdfEncryptionUtility;
+    private PDFUtils mPDFUtils;
+    private Context mContext;
+    private String mDecryptedPath;
 
-    public ExtractImages(String mPath, ExtractImagesListener mExtractImagesListener) {
+    public ExtractImages(Context context, String mPath,
+                         ExtractImagesListener mExtractImagesListener) {
         this.mPath = mPath;
         this.mExtractImagesListener = mExtractImagesListener;
         mOutputFilePaths = new ArrayList<>();
+        this.mContext = context;
+    }
+
+    /**
+     * This function checks whether a file is encrypted or not and if it is
+     * then it decrypts it.
+     *
+     * @param path is the path of the file which needs to be checked for encryption.
+     */
+    private void decryptFile(String path) {
+        if (mPDFUtils.isPDFEncrypted(path)) {
```

Comment:
@vyankatesh24 The pdf should be decrypted that is dialog should appear only if it is encrypted. Why is that for every pdf?

Reply:
Dialog should not appear for every PDF.  Only for encrypted ones.

Reply:
Yes that is what I am saying.

Reply:
The logic is according to that only right? It should enter the statements in the if-block if the file is Encrypted.

Reply:
@vyankatesh24 ?